tensorflow/core/ops/math_ops.cc

```diff
@@ -2365,4 +2365,132 @@ output_max: the computed max output.
 
 )doc");
 
+// --------------------------------------------------------------------------
+
+namespace {
+
+Status ParticalReductionShapeFn(InferenceContext* c) {
+  ShapeHandle handle;
+  DimensionHandle dimhandle;
+  // "indices" must have rank 2 and the number of columns must be 2
+  TF_RETURN_IF_ERROR(c->WithRank(c->input(1), 2, &handle));
+  TF_RETURN_IF_ERROR(c->Merge(c->Dim(c->input(1),1), c->MakeDim(2), &dimhandle));
+  // "data" must have rank at least 1
+  TF_RETURN_IF_ERROR(c->WithRankAtLeast(c->input(0), 1, &handle));
+  // shape of output tensor
+  TF_RETURN_IF_ERROR(c->ReplaceDim(handle, 0, c->Dim(c->input(1),0), &handle));
+  c->set_output(0, handle);
+  return Status::OK();
+}
+
+} // namespace
+
+REGISTER_OP("PartialSum")
```

Comment:
Perhaps we should call this family of ops 
PartialReduceSum, PartialReduceProd, PartialReduceMin, PartialReduceMax so it is analogous to reduce_min, reduce_max, ...